tiledb/api/cpp_api_support/test/unit_cppapi_string.cc

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file tiledb/api/cpp_api_support/test/unit_cppapi_string.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines tests for the CAPIString class of the TileDB C++ API.
+ */
+
+#include <catch.hpp>
+
+#include "tiledb/sm/cpp_api/capi_string.h"
+
+using namespace tiledb::impl;
+
+TEST_CASE(
+    "CAPIString: Test constructor with null parameter throws",
+    "[capi_string][null-param]") {
+  REQUIRE_THROWS_AS(CAPIString(nullptr), std::invalid_argument);
+  tiledb_string_t* string = nullptr;
+  REQUIRE_THROWS_AS(CAPIString(&string), std::invalid_argument);
```

Comment:
This should be two test cases. The description does not describe the second test